lua/autorun/server/sv_pvp_movespeed.lua

```diff
@@ -44,43 +44,56 @@ local function playerIsInBuild( ply )
     return !getPlayerPvpMode( ply )
 end
 
-local function movementMultiplier(weaponCount) 
+local function movementMultiplier( weaponCount )
+    if weaponCount < 1 then return 1 end
     return math.Clamp(1 - (1.9^( weaponCount))/100, 0, 1)
 end
 
-local function setSpeed(ply, multiplier) 
-    -- set speed multiplier, 0 - 1
-    ply:SetRunSpeed(math.Clamp(baseRunSpeed*multiplier,minRunSpeed, baseRunSpeed))
-    ply:SetWalkSpeed(math.Clamp(baseWalkSpeed*multiplier,minWalkSpeed, baseWalkSpeed))
+local function setSpeedFromWeaponCount( ply, weaponCount )
+    local multiplier = movementMultiplier( weaponCount )
+
+    local runSpeed = baseRunSpeed * multiplier
+    local walkSpeed = baseWalkSpeed * multiplier
+    ply:SetRunSpeed( math.max( runSpeed, minRunSpeed ) )
+    ply:SetWalkSpeed( math.max( walkSpeed, minWalkSpeed ) )
 end
 
-local function adjustMovementSpeed(ply) 
-    if playerIsInBuild( ply ) then 
-        setSpeed(ply, 1) 
-        return
+local function getWeaponWeight( weapon )
+    -- the weight/significance of a weapon (1 for affecting weight 0 for not affecting weight)
+    if not nonEffectedWeapons[weapon:GetClass()] then
+        return 1
     end
-    
+    return 0
+end
+
+local function getWeaponCount( ply ) 
+    if playerIsInBuild( ply ) then return 0 end
     local weapons = ply:GetWeapons()
-    local wepCount = 0
-    
-    -- count weapons
-    for k, weapon in pairs(weapons) do
-        if nonEffectedWeapons[weapon:GetClass()] == nil then
-            wepCount =  wepCount+1
-        end
+    local weaponCount = 0
+    for _, weapon in pairs( weapons ) do
```

Comment:
In reference to my other comment, maybe something like this:

```lua
for _, weapon in pairs( weapons ) do
    local weaponHasWeight = unaffectedWeapons( weapon ) ~= nil

    if weaponHasWeight then
        weaponCount = weaponCount + getWeaponWeight( weapon )
    end
end
```

And then should it be called `weaponCount` anymore? Maybe `totalWeaponWeight` is more appropriate?